src/main/java/net/minecraftforge/fml/fixers/ModFixes.java

```diff
@@ -0,0 +1,113 @@
+    {
+        if (builders.containsKey(modid))
+        {
+            return builders.get(modid);
+        }
+        ModFixes builder = new ModFixes(modid, version * 10);
+        builders.put(modid, builder);
+        return builder;
+    }
+
+    public static ImmutableMap<String, ModFixes> getBuilders()
+    {
+        return ImmutableMap.copyOf(builders);
+    }
+
+    public static Map<DataFixer, Pair<Integer, Integer>> getFixVersions(NBTTagCompound nbttagcompound, int defaultMC, int defaultForge)
+    {
+        Map<DataFixer, Pair<Integer, Integer>> versions = getFixVersionsNoVanilla(nbttagcompound, defaultForge);
+        int i = nbttagcompound.contains("DataVersion", Constants.NBT.TAG_ANY_NUMERIC) ? nbttagcompound.getInt("DataVersion") : defaultMC;
+        versions.put(DataFixesManager.getDataFixer(), Pair.of(i, DataFixesManager.FIXER_VERSION));
+        return versions;
+    }
+
+    public static Map<DataFixer, Pair<Integer, Integer>> getFixVersionsNoVanilla(NBTTagCompound nbttagcompound, int defaultForge)
+    {
+        Map<DataFixer, Pair<Integer, Integer>> versions = new HashMap<DataFixer, Pair<Integer, Integer>>();
+        if (nbttagcompound.contains("ForgeDataVersions", Constants.NBT.TAG_COMPOUND))
+        {
+            NBTTagCompound vers = nbttagcompound.getCompound("ForgeDataVersions");
+            for (String modid : ModFixes.getBuilders().keySet())
+            {
+                ModFixes fixes = builders.get(modid);
+                int newVersion = fixes.dataVersion;
+                int version = defaultForge == -2492 ? newVersion : defaultForge;
+                if (vers.contains(modid, Constants.NBT.TAG_ANY_NUMERIC))
+                {
+                    version = vers.getInt(modid);
+                }
+                DataFixer fixer = fixes.getBuiltFixer();
+                versions.put(fixer, Pair.of(version, newVersion));
+            }
+        }
+        return versions;
+    }
+
+    public static Map<DataFixer, Pair<Integer, Integer>> getFixVersions(NBTTagCompound nbttagcompound, boolean useLatestOtherwiseN1)
+    {
+        return getFixVersions(nbttagcompound, (useLatestOtherwiseN1 ? DataFixesManager.FIXER_VERSION : -1), (useLatestOtherwiseN1 ? -2492 : -1));
+    }
+
+    public static Map<DataFixer, Pair<Integer, Integer>> getFixVersionsNoVanilla(NBTTagCompound nbttagcompound, boolean useLatestOtherwiseN1)
+    {
+        return getFixVersionsNoVanilla(nbttagcompound, (useLatestOtherwiseN1 ? -2492 : -1));
+    }
+
+    /*
+     * ######################################### Members #########################################
+     */
+
+    public final String mod;
+    public final int dataVersion;
+    private DataFixer fixer;
+
+    private ModFixes(String modid, int dataVersion)
+    {
+        super(dataVersion);
+        this.dataVersion = dataVersion;
+        this.mod = modid;
+    }
+
+    public String getModId()
+    {
+        return mod;
+    }
+
+    public DataFixer getBuiltFixer()
```

Comment:
Generally you don't want your datafixer to have behavior that changes. For each fixer version, it should fix data the exact same way every single time. I'm not really a fan of allowing mod fixers to be rebuilt.